network/kademlia_test.go

```diff
@@ -461,23 +456,43 @@ func TestSuggestPeers(t *testing.T) {
 	tk.checkSuggestPeer("<nil>", 0, false)
 }
 
+//Tests change of saturationDepth returned by suggestPeers
 func TestSuggestPeersSaturationDepthChange(t *testing.T) {
 
 	base := "00000000"
 	tk := newTestKademlia(t, base)
 	tk.On("00100000", "00010000", "10000000", "11000000", "11100000", "01000000", "01100000")
 
 	//Saturation depth is 2
+	if tk.saturationDepth != 2 {
+		t.Fatalf("Saturation depth should be 2, got %d", tk.saturationDepth)
+	}
 	tk.Off("01000000")
 	//Saturation depth should have fallen to 1
 	tk.checkSuggestPeer("01000000", 1, true)
 	tk.On("01000000")
 
 	//Saturation depth is 2 again
+	if tk.saturationDepth != 2 {
+		t.Fatalf("Saturation depth should be 2, got %d", tk.saturationDepth)
+	}
 	tk.Off("10000000")
 	//Saturation depth should have fallen to 0
 	tk.checkSuggestPeer("10000000", 0, true)
+	tk.On("10000000")
+
+	tk.On("10101010")
```

Comment:
I'm not sure if I understand this last part of the test. Perhaps we should brainstorm some neighbor related cases that should be checked:

* closest bin becomes empty, depth value should decrease
* closest bin gets more than minbinsize connections, depth value should not change
* ....anything else?

Reply:
I was just removing a peer from the neighborhood so that the depth changes. I guess this test case is covered by your first proposed case. 
I can't think of more cases. Will add both to the test